aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/AWSAppSyncDeltaSync.java

```diff
@@ -0,0 +1,540 @@
+                runBaseQuery(AppSyncResponseFetchers.CACHE_ONLY);
+                if (deltaSyncOperationFailed ) {
+                    return;
+                }
+
+                //Subscribe
+                if (subscription != null ) {
+                    mode = QUEUING_MODE;
+                    subscribe();
+                    if (deltaSyncOperationFailed) {
+                        return;
+                    }
+                }
+
+                //Check if we need to execute the DeltaQuery or BaseQuery from the network.
+                boolean executeBaseQuery = true;
+                if ( !forceFetch && deltaQuery != null) {
+                    //Check if
+                    long deltaInSeconds = (System.currentTimeMillis() - (lastRunTimeInMilliSeconds - 2000)) / 1000;
+                    Log.v(TAG, "Delta Sync: Time since last sync [" + deltaInSeconds + "] seconds");
+                    if (deltaInSeconds < baseRefreshIntervalInSeconds) {
+                        Log.v(TAG, "The last baseQuery from NETWORK was executed less than [" + baseRefreshIntervalInSeconds + "] seconds ago. Will run DeltaQuery from network");
+                        executeBaseQuery = false;
+                    }
+                    else {
+                        Log.v(TAG, "The last baseQuery from NETWORK run was before [" + baseRefreshIntervalInSeconds + "] seconds. Will run BaseQuery from network");
+                    }
+                }
+                else {
+                    Log.v(TAG, "Will run BaseQuery from network");
+                }
+
+                if (executeBaseQuery) {
+                    runBaseQuery(AppSyncResponseFetchers.NETWORK_ONLY);
+                }
+                else {
+                    runDeltaQuery();
+                }
+
+                //Propagate all messages received so far and put subscription in processing mode
+                synchronized (processingLock) {
+                    Log.v(TAG, "Delta Sync: Delta query completed. Will propagate any queued messages on subscription");
+                    Response r;
+                    while ((r = messageQueue.poll()) != null) {
+                        if (subscriptionCallback != null ) {
+                            Log.v(TAG, "Delta Sync: Propagating queued message on subscription");
+                            subscriptionCallback.onResponse(r);
+                        }
+                    }
+                    Log.d(TAG, "Delta Sync: All queued messages propagated. Flipping mode to PROCESSING");
+                    mode = PROCESSING_MODE;
+                }
+            }
+        }).start();
+
+        return id;
+    }
+
+    void cancel() {
+         Log.i(TAG, "Delta Sync: Cancelling Delta Sync operation [" + id + "]");
+        cancelled = true;
+        if (deltaSyncSubscriptionWatcher != null ) {
+            deltaSyncSubscriptionWatcher.cancel();
+        }
+        deltaSyncObjects.remove(id);
+        dbHelper.deleteRecord(id);
```

Comment:
why should we delete the record from the dbHelper?

Reply:
Good point. I was looking at  "cancel" as a terminal state as opposed to a temporary suspension of sync. Will remove